src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1615,21 +1615,21 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
     copy2(rdd2 = rdd2.head(n))
   }
 
-  def insertSA(sig: Type, args: String*): (Type, Inserter) = insertSA(sig, args.toList)
+  def insertSA(sig: Type, args: String*): (TStruct, Inserter) = insertSA(sig, args.toList)
 
-  def insertSA(sig: Type, path: List[String]): (Type, Inserter) = saSignature.insert(sig, path)
+  def insertSA(sig: Type, path: List[String]): (TStruct, Inserter) = saSignature.structInsert(sig, path)
 
-  def insertVA(sig: Type, path: List[String]): (Type, Inserter) = {
-    vaSignature.insert(sig, path)
+  def insertVA(sig: Type, path: List[String]): (TStruct, Inserter) = {
+    vaSignature.structInsert(sig, path)
   }
 
   def insertIntoRow[PC](makePartitionContext: () => PC)(typeToInsert: Type, path: List[String],
     inserter: (PC, RegionValue, RegionValueBuilder) => Unit): MatrixTable = {
     val newRDD2 = rdd2.insert(makePartitionContext)(typeToInsert, path, inserter)
     copy2(rdd2 = newRDD2,
       // don't need to update vSignature, insert can't change the keys
-      vaSignature = newRDD2.typ.rowType.fieldType(2),
-      genotypeSignature = newRDD2.typ.rowType.fieldType(3).asInstanceOf[TArray].elementType)
+      vaSignature = newRDD2.typ.rowType.fieldType(2).asInstanceOf[TStruct],
```

Comment:
`insertIntoRow` is called by `annotateVariantsExpr`.  I don't see where `vds.annotate_variants_expr('va = 5')` generates a nice error message, it looks like this cast will fail.  (Have we abandoned api1 already?)

Reply:
this will fail, and will say something like "you tried to insert a non-struct into a top level schema". I don't particularly care enough to add a good error message here, but can do that if you think it's a good idea.

Reply:
Sorry, I wasn't clear.  This calls through the region value insertion stack, while the error message you have in structInsert (which is a perfectly good error message) is the legacy annotation insertion stack.  I don't see an analogous check/error message for this case so I think my example will fail with a cast exception here.

This is annoying to fix and the interface will eventually disallow it, so I'm going to approve and you can fix as you like (now, or merge and make it irrelevant).